lib/ansible/modules/cloud/vmware/vmware_guest_serial_port.py

```diff
@@ -0,0 +1,469 @@
+        if set(required_params).issubset(backing_info.keys()):
+            backing = serial_port.PipeBackingInfo()
+            backing.pipeName = backing_info['pipe_name']
+            backing.endpoint = backing_info['endpoint']
+        else:
+            self.module.fail_json(msg="Failed to create a new serial port of pipe backing type due to insufficient parameters."
+                                  + "The required parameters are pipe_name and endpoint")
+
+        # since no_rx_loss is an optional argument, so check if the key is present
+        if 'no_rx_loss' in backing_info.keys() and backing_info['no_rx_loss']:
+            backing.noRxLoss = backing_info['no_rx_loss']
+        return backing
+
+    def set_device_backing(self, serial_port, backing_info):
+        """
+        Set the device backing params
+        """
+        required_params = ['device_name']
+        if set(required_params).issubset(backing_info.keys()):
+            backing = serial_port.DeviceBackingInfo()
+            backing.deviceName = backing_info['device_name']
+        else:
+            self.module.fail_json(msg="Failed to create a new serial port of device backing type due to insufficient parameters."
+                                  + "The required parameters are device_name")
+        return backing
+
+    def set_file_backing(self, serial_port, backing_info):
+        """
+        Set the file backing params
+        """
+        required_params = ['file_path']
+        if set(required_params).issubset(backing_info.keys()):
+            backing = serial_port.FileBackingInfo()
+            backing.fileName = backing_info['file_path']
+        else:
+            self.module.fail_json(msg="Failed to create a new serial port of file backing type due to insufficient parameters."
+                                  + "The required parameters are file_path")
+        return backing
+
+    def get_backing_info(self, serial_port, backing, backing_type):
+        """
+        Returns the call to the appropriate backing function based on the backing type
+        """
+        switcher = {
+            "network": self.set_network_backing,
+            "pipe": self.set_pipe_backing,
+            "device": self.set_device_backing,
+            "file": self.set_file_backing
+        }
+        backing_func = switcher.get(backing_type, "Invalid Backing Info")
+        return backing_func(serial_port, backing)
+
+    def create_serial_port(self, backing):
+        """
+        Create a new serial port
+        """
+        serial_spec = vim.vm.device.VirtualDeviceSpec()
+        serial_port = vim.vm.device.VirtualSerialPort()
+        serial_port.yieldOnPoll = backing['yield_on_poll'] if 'yield_on_poll' in backing.keys() else True
+        serial_port.backing = self.get_backing_info(serial_port, backing, backing['type'])
+        serial_spec.device = serial_port
+        return serial_spec
+
+
+def get_serial_port(vm_obj, backing):
```

Comment:
Hi, is there any reason that put these two functions outside the class? Thanks.

Reply:
@Tomorrow9 because there were no references to class members in this function, thought it can be outside the class.